### DISCUSSION TITLE
LInk to file then create it with a template

### DISCUSSION THREAD

**Discussion by fallacius** [SilentVoid13/Templater#666] [Help]
Hello!

I am currently trying to make a recipe collection with has experiences associated with it. In the recipe file, I want to be able to insert a template that links to a new experience file in the recipe file (File name of recipe + date) and then create the experience file with "cookingExperience" template.

This is my Recipe to Experience template:
```js
<%* title = tp.file.title
date = tp.date.now("YYYY-MM-DD ddd")
newTitle = title + " " + date 
tR += "[" + "[" + newTitle + "]]"
path = await this.app.vault.getAbstractFileByPath("300 Learning/370 Food/371 Recipes/Experiences")
template = await this.app.vault.getAbstractFileByPath("999 Templates/Personal/cooking/cookingExperience.md")
await tp.file.create_new(template, newTitle, true, path) -%> 
```

And this is my recipe template, it is relatively simple:
```md
---
tags: recipe
cssclass: recipe
---
# <% tp.file.title %>
## 💬 Description:

## 🛠️ Tools:

...
```

I am not quite sure why the link is prepended onto the newly created experiences file instead of the recipe file when I insert the Cooking To Experience template in the recipe file. 

Thank you in advance!

**Comment by welpdx** [comment]
oh lord have mercy

**Reply by welpdx** [reply]
this is an hell of an issue XD

**Reply by AB1908** [reply]
Were you able to repro? I think I saw this user come by on Discord and I had nothing to offer them then, sadly.

**Reply by welpdx** [reply]
Edit: 
Yes I was able to reproduce the bug on a new vault. It seemed like the tR was returned only after the last few await was executed. 
It has something to do with `active_leaf.openFile()` executing before tR can be returned and inserted in the first file.

<details>
           <summary style="font-size:20">Repro</summary>
           <img src="https://i.imgur.com/EBcW18h.gif" alt="demo" />
</details>

**Comment by welpdx** [comment]
I suggest you turn off open_new for `tp.file.create_new()`

Your script works as intended if you do: `await tp.file.create_new(template, newTitle, false, path)`
